### DISCUSSION TITLE
Can I change which router an end-device is connected to

### DISCUSSION THREAD

**Discussion by ghost** [Koenkk/zigbee2mqtt#12882] [General]
Hi,

It's not the end of the world, but I have many examples where a motion sensor or switch end-device is not linked to the bulb router that is in the same room so it has a lower link quality.

Is there a way to change this?  ...is it even worth doing?

**Comment by Koenkk** [comment]
No this is not possible in Zigbee, you cannot manually assign parents to certain devices.

**Comment by ghost** [comment]
cheers.

Does this mean there is no 're-optimise' functionality that allows the end-devices to re-select the best strength parent/router?

**Reply by Koenkk** [reply]
They should do that already (it is the responsibility of the device to do this).

**Comment by ghost** [comment]
I'm not sure if it is doing this.  I have several Hue sensors/switches that are not linked to the nearest (and one would assume strongest) router/parent ion the same room but, instead, ones either outside the room, through an exterior wall or even down the corridor and round the corner.

**Comment by ut1545** [comment]
AFAIK, in the frontend you can select the router that you want to join. If you select ALL, the device is joined to the "best" router or to coordinator but if you select a specific router, the device is joined to it and remains connected. I haven't tested but I think that if the router that the device is connected to goes offline, the device is reconnected to a new router, if possible, and remains connected until this router become unavailable.
BTW, I don't have a deep knowledge of zigbee protocols so, this is based on some observations and tests.

**Reply by ghost** [reply]
cheers!
